syntax = "proto3";
package api;

message Message {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
}

message TaskData {
  string user_id = 1 [json_name = "userId"];
  string task_id = 2 [json_name = "taskId"];
  string task_type = 3 [json_name = "taskType"];
  string task_stage = 4 [json_name = "taskStage"];
  int32 status = 5 [json_name = "status"];
  int32 crt_retry_num = 6 [json_name = "crtRetryNum"];
  int32 max_retry_num = 7 [json_name = "maxRetryNum"];
  int32 max_retry_interval = 8 [json_name = "maxRetryInterval"];
  string schedule_log = 9 [json_name = "scheduleLog"];
  string task_context = 10 [json_name = "context"];
  int32 order_time = 11 [json_name = "orderTime"];
  string create_time = 12 [json_name = "createTime"];
  string modify_time = 13 [json_name = "modifyTime"];
  string id = 14 [json_name = "Id"];
}

message TaskScheduleCfg {
  string task_type = 1 [json_name = "taskType"];
  int32 schedule_limit = 2 [json_name = "scheduleLimit"];
  int32 schedule_interval = 3 [json_name = "scheduleInterval"];
  int32 max_processing_time = 4 [json_name = "maxProcessingTime"];
  int32 max_retry_num = 5 [json_name = "maxRetryNum"];
  int32 max_retry_interval = 6 [json_name = "maxRetryInterval"];
  string create_time = 7 [json_name = "createTime"];
  string modify_time = 8 [json_name = "modifyTime"];
}

message CreateTaskReq {
  TaskData taskdata = 1 [json_name = "taskData"];
}

message CreateTaskRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  string task_id = 3 [json_name = "task_id"];
}

message GetTaskListReq {
  string task_type = 1 [json_name = "task_type"];
  int32 status = 2 [json_name = "status"];
  int32 limit = 3 [json_name = "limit"];
}

message GetTaskListRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  repeated TaskData task_list = 3 [json_name = "taskList"];
}

message HoldTasksReq {
  string task_type = 1 [json_name = "taskType"];
  int32 limit = 2 [json_name = "limit"];
}

message HoldTasksRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  repeated TaskData task_list = 3 [json_name = "taskList"];
}

message GetTaskReq {
  string task_id = 1 [json_name = "taskId"];
}

message GetTaskRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  TaskData task_data = 3 [json_name = "task_data"];
}

message GetTaskCountByStatusReq {
  string task_type = 1 [json_name = "taskType"];
  int32 status = 2 [json_name = "status"];
}

message GetTaskCountByStatusRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  int32 count = 3 [json_name = "count"];
}

message GetTaskScheduleCfgListReq {
}

message GetTaskScheduleCfgListRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
  repeated TaskScheduleCfg task_schedule_cfg = 3 [json_name = "scheduleCfgList"];
}

message SetTaskStatusReq {
  string task_id = 1 [json_name = "taskId"];
  int32 status = 2 [json_name = "status"];
  bool no_modify_time = 3 [json_name = "noModifyTime"];
}

message SetTaskStatusRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
}

message SetTaskReq {
  string user_id = 1 [json_name = "userId"];
  string task_id = 2 [json_name = "taskId"];
  string task_type = 3 [json_name = "taskType"];
  string task_stage = 4 [json_name = "taskStage"];
  int32 status = 5 [json_name = "status"];
  int32 crt_retry_num = 6 [json_name = "crtRetryNum"];
  int32 max_retry_num = 7 [json_name = "maxRetryNum"];
  int32 max_retry_interval = 8 [json_name = "maxRetryInterval"];
  string schedule_log = 9 [json_name = "scheduleLog"];
  string task_context = 10 [json_name = "context"];
  int32 order_time = 11 [json_name = "orderTime"];
}

message SetTaskRsp {
  int32 code = 1 [json_name = "code"];
  string msg = 2 [json_name = "msg"];
}

message ScheduleData {
  string TraceID = 1 [json_name = "traceId"];
  string ErrorMsg = 2 [json_name = "errMsg"];
  string cost = 3 [json_name = "cost"];
}